.devcontainer/prebuild/.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,11 @@
+{
+    "build": {
+        "args": {
+            "PYTHON_VERSION": "${localEnv:PYTHON_VERSION}"
```

Comment:
imo, it is not necessary to prepare images based on python version. after all, image construction and storage require costs.

maybe we can build a common devcontainer capable of switching between different versions but it is overall fine for this.

Reply:
afaik, mamba/conda from <https://containers.dev/features> can be the option to build a common and unique development environment image.
- micromamba: <https://github.com/mamba-org/devcontainer-features/tree/main/src/micromamba>
- conda: <https://github.com/rocker-org/devcontainer-features/tree/main/src/miniforge>

Reply:
Looks cool. I will take a look then.

Reply:
Hi @msclock, I just come back from vacation. Sorry for the late reply.

mamba/conda should be a choice to manage different Python versions. [pdm](https://pdm-project.org/latest/reference/cli/#use), [pipenv](https://pipenv.pypa.io/en/latest/specifiers.html#specifying-versions-of-python) and [poetry](https://python-poetry.org/docs/managing-environments/#switching-between-environments) also have similar functionality to achieve the same goal. So we can narrow down the problem to whether we should maintain an image with different Python versions. Considering the similar scenario for PyTorch, generally, we will just use [the NGC image](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch) for ease. In this way, I would prefer to keep the same pattern, a.k.a, different images for different versions. I created an issue https://github.com/serious-scaffold/ss-python/issues/265 to track this, any further suggestions or comments are welcome.

BTW, I prefer not to use devcontainer features (and templates), as I did not see any advantage when compared with maintaining a more general multi-stage Dockerfile combined with [prebuilds](https://containers.dev/guide/prebuild). Since all the default Python images are based on [the same base image](https://github.com/docker-library/python/blob/master/3.10/bookworm/Dockerfile#L7), both the image construction time and storage cost should be acceptable.

Reply:
@huxuan hi, thanks for your reply. because of the current development in an alpha stage, what i expect is to suggest a more general and simple development environment. about
 the deployment based on container maybe is not neccesary. Moreover, the built app in the scaffold is a simple cmdline tool. So initially, i think using common features used in a development environment should be ok.